lib/Driver/Compilation.cpp

```diff
@@ -421,14 +424,26 @@ int Compilation::performJobsImpl() {
   }
 
   int Result = EXIT_SUCCESS;
+  llvm::TimerGroup DriverTimerGroup("Driver Time Compilation");
+  std::map<std::string, std::unique_ptr<llvm::Timer>> DriverTimers;
```

Comment:
I think this is a perfectly good approach, except the map should be from `Job *` to timer instead of command-line-string to timer. (At which point I suggest using an `llvm::DenseMap`, which is a hash map. See the [LLVM Programmer's Manual](http://llvm.org/docs/ProgrammersManual.html#picking-the-right-data-structure-for-a-task) for more nifty data structures.)